core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -216,12 +222,22 @@ public void onNext(HttpObject o) {
     @Override
     public void onError(Throwable cause) {
         if (cause instanceof HttpResponseException) {
-            failAndRespond(cause, ((HttpResponseException) cause).httpStatus(), Http2Error.CANCEL);
+            // Timeout may be occurred because calling failAndRespond is postponed.
```

Comment:
- `Timeout may occur when the aggregation of the error response take long.`?
- How about adding this comment as well? - `If timeout occurs, respond with 500 Internal Server Error.`
  - How about reusing `HttpResponse.status()` instead?